library/common/extensions/cert_validator/platform_bridge/c_types.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "library/common/types/c_types.h"
+
+// NOLINT(namespace-envoy)
+
+/**
+ * Certification validation binary result with corresponding boring SSL alert
+ * and error details if the result indicates failure.
+ */
+typedef struct {
+  envoy_status_t result;
+  uint8_t tls_alert;
+  const char* error_details;
+} envoy_cert_validation_result;
+
+#ifdef __cplusplus
+extern "C" { // function pointers
+#endif
+
+/**
+ * Function signature for calling into platform APIs to validate certificates.
+ */
+typedef envoy_cert_validation_result (*envoy_validate_cert_f)(const envoy_data* certs, uint8_t size,
+                                                              const char* host_name);
+
+/**
+ * Function signature for calling into platform APIs to clean up after validation completion.
+ */
+typedef void (*envoy_validation_done_f)();
+
+#ifdef __cplusplus
+} // function pointers
+#endif
+
+/**
+ * A bag of function pointers to be registered in the platform registry.
+ */
+typedef struct {
+  envoy_validate_cert_f validate_cert;
+  envoy_validation_done_f validation_done;
```

Comment:
nit: Based on my reading of the code it looks like this is exclusively for cleanup. If so, elsewhere we've adopted a pattern where we'd call this something like `release_validator`.